analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,357 @@
+---
+title: "Find most focal recurrent copy number units"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: Chante Bethell for ALSF CCDL
+date: 2020
+params:
+  is_ci: FALSE
+---
+
+This notebook defines the most focal recurrent copy number units by removing focal changes that are within entire chromosome arm losses and gains.
+_Most focal_ here meaning:
+
+- If a chromosome arm is not clearly defined as a gain or loss (and is callable) we look to define the cytoband level status
+- If a cytoband is not clearly defined as a gain or loss (and is callable) we then look to define the gene level status
+
+## Usage
+
+This notebook is intended to be run from the command line with the following (assumes you are in the root directory of the repository):
+
+```
+Rscript -e "rmarkdown::render('analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+### Libraries and functions
+
+```{r}
+library(tidyverse)
+```
+
+### Custom Function
+
+```{r}
+status_majority_caller <- function(status_df,
+                                   region_variable) {
+  # Given a data.frame with cytoband/gene level copy number status data,
+  # find the dominant status of the cytoband/gene region by calculating
+  # the percentages of the region that each call represents.
+  #
+  # Args:
+  #   status_df: data.frame with cytoband/gene level copy number status data
+  #   region_variable: string of the column name/region to calculate copy number
+  #                    status percentages for
+  # Return:
+  #   status_count: data.frame with percentage values for each unique status in
+  #                 each unique region (arm/cytoband/gene) and the dominant
+  #                 status for that region
+
+  region_sym <- rlang::sym(region_variable)
+
+  if (region_variable == "chromosome_arm") {
+    status_count <- status_df %>%
+      count(!!region_sym, dominant_status) %>%
```

Comment:
Am I understanding this that you want this to use `dominant_status` for `region_variable = chromosome_arm`, but should use `status` if `region_variable = cytoband` or `gene_symbol`?

Reply:
Yes, that is correct. The data.frame that `cytoband` and `gene_symbol` data are retrieved from has the column name `status` and not `dominant_status`.

However, now thinking about how this can be refactored, I can just rename the data.frame column name to be the same across data.frames.

Reply:
If you go with the renaming route, I would keep that renaming just to the function, but a preferred solution might be to just specify another tidyeval so that you don't lose this column name information.

Reply:
Ahh okay, I'll give the second solution mentioned here a try first.